src/components/common/Callout.scss

```diff
@@ -1,35 +1,127 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import './color';
+@import './radius';
+@import './utils';
+
 .Callout {
   display: flex;
-  position: relative;
-
-  &::before {
-    background-color: var(--mdc-theme-secondary);
-    bottom: 0;
-    content: '';
-    left: 0;
-    opacity: 0.08;
-    position: absolute;
-    right: 0;
-    top: 0;
-  }
 }
 
 .Callout-body {
-  padding: 12px 0 12px 8px;
-  margin-right: 24px;
+  padding: 12px 24px;
+}
 
-  .rmwc-icon {
-    vertical-align: middle;
-    margin-left: 18px;
-    margin-right: 8px;
-  }
+.Callout-message {
+  display: flex;
+  align-items: stretch;
+}
+
+.Callout-icon {
+  margin-right: 12px;
 }
 
 .Callout-actions {
   align-items: center;
   display: flex;
-  flex: 1 1 auto;
   flex-direction: row;
+  flex: 1 1 auto;
   justify-content: flex-end;
-  padding-right: 8px;
+  padding-right: 16px;
+}
+
+// situational overrides
+.mdc-card > .Callout:first-child {
+  border-top-left-radius: $BORDER_RADIUS_CARD;
+  border-top-right-radius: $BORDER_RADIUS_CARD;
+}
+
+.mdc-card > .Callout:last-child {
+  border-bottom-left-radius: $BORDER_RADIUS_CARD;
+  border-bottom-right-radius: $BORDER_RADIUS_CARD;
+}
+
+// THEMES
+$msg-types: (
```

Comment:
Thoughts on using functions instead of list + loop?

Reply:
If were going to re-asses this, lets keep as is. it is likely going to be deleted.